### PR TITLE
Spiders don't automatically grant an antag datum

### LINKED ISSUE #77501: Transforming into a Giant Spider gives you a persistent antag datum
## Reproduction:

The way I found out this bug is I took the dna of a spider and became it to test if I could become a spider then after turning human again the spider tab stayed on my character but steps to reproduce:
1. Get the polymorph inverter and insert the bioscrambler core.
2. Grow a spider from a gold slime core or find a corpse.
3. Become said spider then turn human again once the cooldown is done.
Bug

### PR DESCRIPTION
**Pull request by Jacquerel**
## About The Pull Request

Fixes #77501

Spider egg ghost role spawners grant the spider antag datum, rather than the act of being a spider.
This means that gold core, mapstart, polymorph belt, and other spiders will not have an antagonist datum.

While doing this I also made a new abstract `mob/living/basic/spider` type which all three kinds of spider life stage (`spiderling`, `young`, `giant`) extend from, because there was a gross amount of copied code. Now there isn't.

Also the Flesh Spider and Event Midwife eggs now simply hatch adult spiders instead of child ones.
This is because there is no reason for either of these to have a two minute wait time before they get going. Midwife spiders spawned by the event should just start spidering immediately, and Flesh Spiders are made by changelings and shouldn't be effected by measures introduced to balance the spider gamemode.
Eggs which are laid during a round and _can_ hatch into midwife spiders still hatch baby spiders.

Also I swapped some white pixels on the animation of the ambush spiderling for a different colour because they looked bad.

## Why It's Good For The Game

While the policy is always "if you turn yourself into something, you're not an antagonist" the presence of the antag datum still confuses people. Plus that code was gross and I didn't like it.

## Changelog

:cl:
fix: Giant Spiders only have an antag datum if created by the round event.
balance: Flesh spider eggs hatch into adult spiders instead of baby spiders.
balance: The eggs spawned by the start of the spider infestation event hatch into adult Midwife spiders instead of baby ones.
/:cl:

### CODE REVIEW

**Comment by optimumtact** [comment]
policy cannot depend on the existence or not of datums, this PR is wrongheaded in the problem it seeks to fix.

**Comment by LemonInTheDark** [comment]
agree with the above, if admins are doing stuff based off the existence of an antag datum then that is an issue with the admin team not us

**Comment by Jacquerel** [comment]
> policy cannot depend on the existence or not of datums, this PR is wrongheaded in the problem it seeks to fix.

The policy is always "if you turn yourself into a mob, you're not an antagonist, even if it says you are" but reducing the confusion is good.
Also the code about whether or not to add the datum in the first place was a mess, and I don't think it _should_ be granting the datum anyway under these other circumstances.

**Comment by TheVekter** [comment]
> This wasn't requested by admins (as far as I know they already know how to deal with this because it has historically worked this way, and so do xenos)

This is correct; this topic comes up from time to time and some of us older admins are always very adamant that "having an antag datum" isn't directly indicative of antag status.

This is less about codifying admin mistakes and more about making things more sane for players.

File: code/modules/mob/living/basic/space_fauna/spider/young_spider/young_spider_subtypes.dm [hunk 1]
```diff
@@ -116,30 +116,30 @@
...
+/mob/living/basic/spider/growing/young/tangle/Initialize(mapload)
 	. = ..()
 	AddComponent(/datum/component/healing_touch,\
 		heal_brute = 10,\
 		heal_burn = 10,\
 		heal_time = 3 SECONDS,\
 		self_targetting = HEALING_TOUCH_SELF_ONLY,\
 		interaction_key = DOAFTER_SOURCE_SPIDER,\
-		valid_targets_typecache = typecacheof(list(/mob/living/basic/giant_spider/tangle)),\
+		valid_targets_typecache = typecacheof(list(/mob/living/basic/spider/giant/tangle)),\
```
Comment (Comxy) [inline code/modules/mob/living/basic/space_fauna/spider/young_spider/young_spider_subtypes.dm:127]: Can you change this to /mob/living/basic/spider/growing/young/tangle? Then I can close my fix pr

**Pull request merged**